machine-learning/02-openpbta_consensus_clustering.Rmd

```diff
@@ -0,0 +1,378 @@
+
+## Low-dimensional representation
+
+Let's see if low-dimensional representations agree with our other unsupervised results.
+First, let's find the high variance genes; we'll perform dimension reduction on those genes only.
+
+```{r gene_variance}
+# Calculate variance
+gene_variance <- matrixStats::rowVars(rnaseq_mat)
+# Find the value that we'll use as a threshold to filter the top 5%
+variance_threshold <- quantile(gene_variance, 0.95)
+# Row indices of high variance genes
+high_variance_index <- which(gene_variance > variance_threshold)
+```
+
+### UMAP
+
+First, let's use UMAP.
+
+```{r run_umap}
+# Set seed for reproducible UMAP results
+set.seed(2020)
+# umap() expects features (genes) to be columns, so we have to use t()
+umap_results <- umap::umap(t(rnaseq_mat[high_variance_index, ]))
+```
+
+The UMAP coordinates are in the `layout` element of the list returned by `umap::umap()`.
+
+```{r make_umap_df, live = TRUE}
+# Make a data frame of the layout results and join with molecular subtype 
+# data frame
+umap_plot_df <- data.frame(umap_results$layout) %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  inner_join(mb_molecular_subtype_df)
+```
+
+Let's make a scatter plot and color our samples by the subtype labels.
+
+```{r umap_scatter}
+umap_plot_df %>%
+  ggplot(aes(x = X1, 
+             y = X2,
+             color = molecular_subtype)) +
+  geom_point() +
+  colorblindr::scale_color_OkabeIto() +
+  theme_bw() +
+  xlab("UMAP1") +
+  ylab("UMAP2")
+```
+
+### PCA 
+
+Now, we'll briefly show you how to use built-in functions for PCA on any matrix that's not in a specialized object for some kind of genomic data.
+
+```{r run_pca}
+# Like umap(), prcomp() expects the features to be columns
+pca_results <- prcomp(t(rnaseq_mat[high_variance_index, ]), 
+                      scale = TRUE)
+# The low-dimensional representation is returned in x
+pca_results$x[1:6, 1:6]
+```
+
+Let's get the first 10 PCs ready for plotting.
+
+```{r pca_df, live = TRUE}
+pca_plot_df <- data.frame(pca_results$x[, 1:10]) %>%
+  tibble::rownames_to_column("Kids_First_Biospecimen_ID") %>%
+  inner_join(mb_molecular_subtype_df)
+```
+
+And now make a scatterplot of PC1 and PC2.
+
+```{r pca_scatter_1, live = TRUE}
+pca_plot_df %>%
+  ggplot(aes(x = PC1, 
+             y = PC2,
+             color = molecular_subtype)) +
+  geom_point() +
+  colorblindr::scale_color_OkabeIto() +
+  theme_bw() +
+  xlab("PC1") +
+  ylab("PC2")
+```
+
+What about PC2 and PC3?
+
+```{r pca_scatter_2, live = TRUE}
+pca_plot_df %>%
+  ggplot(aes(x = PC2, 
+             y = PC3,
+             color = molecular_subtype)) +
+  geom_point() +
+  colorblindr::scale_color_OkabeIto() +
+  theme_bw() +
+  xlab("PC2") +
+  ylab("PC3")
+```
+
+`summary()` will report the proportion of variance explained by each principal component.
+By accessing the `importance` element with `<summary results>$importance`, we can use indexing to only look at the first 10 PCs.
```

Comment:
I think this is a good thing to throw in, but it seems oddly placed here. Maybe some more context about why we care would help? How would you use this information to inform plotting?